apiserver/params/crossmodel.go

```diff
@@ -22,3 +22,44 @@ type CrossModelOffer struct {
 type CrossModelOffers struct {
 	Offers []CrossModelOffer `json:"offers"`
 }
+
+// RemoteEndpoint holds information about remote endpoint.
+type RemoteEndpoint struct {
+	// Name is exported relation name.
+	Name string `json:"name"`
+
+	// Interface is exported relation interface.
+	Interface string `json:"interface"`
+
+	// Role is exported relation role.
+	Role string `json:"role"`
+}
+
+// RemoteServiceInfoResult holds remote service information for a URL or
+// a related error.
+type RemoteServiceInfoResult struct {
+	// Result contains remote service information.
+	Result RemoteServiceInfo `json:"result"`
```

Comment:
as before, use a pointer an ,omitempty so we can elide in the case of an error. lack of precedence is not a good argument, because existing ones have just been cargo-culted